toppra/dracula/__init__.py

```diff
@@ -162,16 +144,58 @@ def lims_obeyed(self, traj, raise_2nd_order):
                 return False
         return True
 
+    def _estimate_path(self, t_sum_multiplier, pc_vel, pc_acc):
+        """Estimate path length from reduced v/a limits.
+
+        This sets up the path only after pc_vel and pc_acc
+        have been established.
+        """
+        # check for duplicates
+        self.min_pair_dist, t_sum = _check_waypts(
+            self.waypts, pc_vel.vlim, pc_acc.alim
+        )
+        if self.min_pair_dist < JNT_DIST_EPS:  # issue a warning and try anyway
+            logger.warning(
+                "Duplicates found in input waypoints. This is not recommended,"
+                " especially for the beginning and the end of the trajectory. "
+                "Toppra might throw a controllability exception. "
+                "Attempting to optimise trajectory anyway..."
+            )
+        # initial x for toppra's path, essentially normalised time on x axis
+        # rescale by given speed limits.
+        # only applies to ParametrizeSpline.
+        self.path_length_limit = 100 * t_sum  # empirical magic number
+        # t_sum is the minimum time required to visit all given waypoints.
+        # toppra generally needs a smaller number for controllabiility.
```

Comment:
You mean it needs more time than the minimum time, right?

Reply:
it needs a smaller path length than the fastest possible `t_sum` for better controllability.

Reply:
Please correct me if I am wrong here: my understanding is that `t_sum` is the minumum time needed by the joint with longest path length to complete its trajectory, assuming instant acceleration/deceleration between 0 and max velocity. Is this correct?